projects/packages/videopress/src/class-ajax.php

```diff
@@ -115,15 +116,144 @@ public function wp_ajax_videopress_get_playback_jwt() {
+		$restriction_details = array(
+			'version'              => '1',
+			'provider'             => 'auth',
+			'title'                => __( 'Unauthorized', 'jetpack-videopress-pkg' ),
+			'unauthorized_message' => __( 'Unauthorized', 'jetpack-videopress-pkg' ),
+			'can_access'           => $default_can_access,
+		);
+
+		return $restriction_details;
+	}
+
+	/**
+	 * Filers restriction details.
+	 *
+	 * @param array $video_restriction_details The restriction details.
+	 *
+	 * @return array
+	 */
+	private function filter_video_restriction_details( array $video_restriction_details ) {
+		return (array) apply_filters( 'videopress_video_restriction_details', $video_restriction_details );
+	}
+
+	/**
+	 * Returns the default restriction_details for a video.
+	 *
+	 * @param bool $default_can_access The default auth.
+	 *
+	 * @return array
+	 **/
+	private function get_subscriber_only_restriction_details( $default_can_access = false ) {
+		return array(
+			'provider'             => 'jetpack_memberships',
+			'title'                => __( 'This video is subscriber-only', 'jetpack-videopress-pkg' ),
+			'unauthorized_message' => __( 'You need to be subscribed to view this video', 'jetpack-videopress-pkg' ),
+			'can_access'           => $default_can_access,
+		);
+	}
+
+	/**
+	 * Determines if Jetpack Memberships are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_memberships_available() {
+		return class_exists( '\Jetpack_Memberships' );
+	}
+
+	/**
+	 * Determines if Jetpack Subscriptions are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_subscriptions_available() {
+		if ( function_exists( '\Automattic\Jetpack\Extensions\Premium_Content\subscription_service' ) ) {
+			return true;
+		}
+
+		if ( ! defined( 'JETPACK__PLUGIN_DIR' ) ) {
+			return false;
+		}
+
+		$subscription_service_file_path = JETPACK__PLUGIN_DIR . 'extensions/blocks/premium-content/_inc/subscription-service/include.php';
+		if ( ! file_exists( $subscription_service_file_path ) ) {
+			return false;
+		}
+
+		require_once $subscription_service_file_path;
+
+		return function_exists( '\Automattic\Jetpack\Extensions\Premium_Content\subscription_service' );
+	}
+
+	/**
+	 * Determines if the current user can access restricted content and builds the restriction_details array.
+	 *
+	 * @param string $guid the video guid.
+	 * @param int    $embedded_post_id the post id.
+	 * @param int    $selected_plan_id the selected plan id.
+	 *
+	 * @return array
+	 */
+	private function build_restriction_details( $guid, $embedded_post_id, $selected_plan_id ) {
+		global $post;
+		$post_to_check = get_post( $embedded_post_id );
+
+		if ( empty( $post_to_check ) ) {
+			$restriction_details = $this->default_video_restriction_details( false );
+			return $this->filter_video_restriction_details( $restriction_details );
+		}
+
+		// phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited
+		$post = $post_to_check;
```

Comment:
Should it somehow reset the postdata afterwards ?
Looking up the documentation :
`$post = setup_postdata( $post_to_check );`
And later on
`wp_reset_postdata();`

Reply:
Since this is a wp_ajax call, I think it is OK as the request is specifically about the specific post_id. On the other hand it doesn't hurt to cleanup.